src/dap/config.ts

```diff
@@ -2,28 +2,25 @@ import * as path from 'path';
 import * as vscode from 'vscode';
 import { getExtensionSetting } from '../utils/settings';
 
+export const DebugEditorContentsCommandId = 'docker.debug.editorContents';
+
 class DebugAdapterExecutableFactory
   implements vscode.DebugAdapterDescriptorFactory
 {
   createDebugAdapterDescriptor(
     session: vscode.DebugSession,
   ): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
-    const parent = path.dirname(session.configuration.dockerfile);
-    return new vscode.DebugAdapterExecutable(
-      'docker',
-      [
-        'buildx',
-        'dap',
-        'build',
-        '-f',
-        session.configuration.dockerfile,
-        parent,
-      ],
-      {
-        cwd: parent,
-        env: { BUILDX_EXPERIMENTAL: '1' },
-      },
-    );
+    var args = ['buildx', 'dap', 'build'];
+    if (session.configuration?.args) {
+      args = args.concat(session.configuration?.args);
+    }
+
+    const options = {
+      env: { BUILDX_EXPERIMENTAL: '1' },
```

Comment:
- Heads up; this will no longer be needed in the next release https://github.com/docker/buildx/pull/3241